org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/launching/ScalaDebuggerTab.scala

```diff
@@ -44,19 +65,27 @@ private class GeneralSettingsBlock(val tab: ScalaDebuggerTab) {
    * @param configuration the configuration to get attributes from
    * @throws CoreException if an error occurs getting an attribute
    */
-  def initializeFrom(configuration: ILaunchConfiguration): Unit =
+  def initializeFrom(configuration: ILaunchConfiguration): Unit = {
     globalAsyncDebuggerSwitchButton.setSelection(configuration.getAttribute(LaunchWithAsyncDebugger, false))
+    import scala.collection.JavaConverters._
+    stepOutExcludePackages.setText(configuration.getAttribute(StepOutExcludePkgsOrClasses,
+      List.empty[String].asJava).asScala.mkString(DataDelimiter))
+  }
 
   /**
    * Sets attributes on the configuration based on the current state of the UI elements
    * @param configuration configuration to modify
    */
-  def performApply(configuration: ILaunchConfigurationWorkingCopy): Unit =
+  def performApply(configuration: ILaunchConfigurationWorkingCopy): Unit = {
     if (globalAsyncDebuggerSwitchButton.getSelection()) {
       configuration.setAttribute(LaunchWithAsyncDebugger, true)
     } else {
       configuration.removeAttribute(LaunchWithAsyncDebugger)
     }
+    import scala.collection.JavaConverters._
+    configuration.setAttribute(StepOutExcludePkgsOrClasses,
+      stepOutExcludePackages.getText.split(DataDelimiter).filterNot { _.isEmpty }.toList.asJava)
```

Comment:
not sure, this is method on String, let me check and if there is `nonEmpty` I replace it (is more natural)